Absolute path to libexpat in generated exiv2Config.cmake

**Describe the bug**
The generated and packaged share/cmake/exiv2Config.cmake contains an absolute path, e.g.

```
# Create imported target exiv2-xmp
add_library(exiv2-xmp STATIC IMPORTED)

set_target_properties(exiv2-xmp PROPERTIES
  INTERFACE_LINK_LIBRARIES "D:/a/_temp/msys/msys64/mingw64/lib/libexpat.dll.a"
)
```

**To Reproduce**
From msys2 terminal, run `makepkg-mingw` on the mingw-w64-exiv2/PKGBUILD and observe the telling

```
==> WARNING: Package contains reference to $(cygpath -m /)
````

near the end of the output.

**Expected behavior**
No warning during build and no absolute paths in shipped & installed package.

**Desktop (please complete the following information):**
 - Version 0.27.3
 - OS: Windows
 - Compiler & Version: gcc 10.2 (mingw-w64)

I have tried different things. I'll try to give a summary here:

When removing the installation of the **exiv2-xmp** target (which for me, it should not be needed, since it is an implementation detail) as I did in #1364, the absolute paths are removed from the cmake config file `share/cmake/exiv2Config.cmake`. However, when trying to configure the project with `-DBUILD_SHARED_LIBS=OFF` (to generate exiv2lib as a STATIC library), CMake says that it needs to export information about the target **exiv2-xmp**:

```
cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DCMAKE_INSTALL_PREFIX=install ../
...
CMake Error: install(EXPORT "exiv2Config" ...) includes target "exiv2lib" which requires target "exiv2-xmp" that is not in any export set.
-- Generating done
```

If we leave the code as it is, the CMake configuration file will contain information about the **exiv2-xmp** target, which is depending on LIBEXPAT (either its STATIC or SHARED version, but always as a absolute path)

```
# Create imported target exiv2-xmp
add_library(exiv2-xmp STATIC IMPORTED)

set_target_properties(exiv2-xmp PROPERTIES
  INTERFACE_LINK_LIBRARIES "/home/luis/.conan/data/expat/2.2.10/_/_/package/2af715f34a7c8c2aeae57b25be0a52c4110dc502/lib/libexpat.so"
)
```

However, I am not sure if this would create a problem for a consumer of **exiv2lib**. That absolute path is associated to the target **exiv2-xmp** and consumers of exiv2 should normally only consume **exiv2lib**. Altough, I understand the issue mentioned by @kmilos , in which he is getting the warning:

```
==> WARNING: Package contains reference to $(cygpath -m /)
```

Just before publishing this comment, I noticed that I can hide the absolute path by using this:

```
target_link_libraries(exiv2-xmp
    PRIVATE 
        $<BUILD_INTERFACE:${EXPAT_LIBRARY}>
)
```

instead of:

```
target_link_libraries(exiv2-xmp
    PRIVATE 
        ${EXPAT_LIBRARY}
)
```

Let's see what the CI jobs have to say about it ...

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -54,7 +54,7 @@ install:
     - set PATH=C:\projects\deps\ninja;%PATH%
     - ninja --version
     - python -m pip install --upgrade pip
-    - pip3.exe install conan==1.24.1
+    - pip3.exe install conan==1.30.2
     - pip3.exe install lxml
     - cd %APPVEYOR_BUILD_FOLDER%
 

--- a/ci/install.sh
+++ b/ci/install.sh
@@ -14,14 +14,14 @@ if [[ "$(uname -s)" == 'Linux' ]]; then
     sudo pip install virtualenv
     virtualenv conan
     source conan/bin/activate
-    pip install conan==1.22.0
+    pip install conan==1.30.2
     pip install codecov
     pip install lxml
 else
     sudo pip3 install virtualenv
     virtualenv conan
     source conan/bin/activate
-    pip3 install conan==1.22.0
+    pip3 install conan==1.30.2
     pip3 install codecov
     pip3 install lxml
 fi

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -144,7 +144,10 @@ set_target_properties( exiv2lib_int PROPERTIES
 )
 
 target_include_directories(exiv2lib_int PRIVATE ${ZLIB_INCLUDE_DIR})
-target_include_directories(exiv2lib PRIVATE ${ZLIB_INCLUDE_DIR})
+target_include_directories(exiv2lib PRIVATE 
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/xmpsdk/include>
+    ${ZLIB_INCLUDE_DIR}
+)
 
 if (EXIV2_ENABLE_XMP)
     target_link_libraries(exiv2lib PRIVATE exiv2-xmp)

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -211,7 +211,7 @@ else()
 endif()
 
 if( EXIV2_ENABLE_PNG )
-    target_link_libraries( exiv2lib PRIVATE ${ZLIB_LIBRARIES} )
+    target_link_libraries( exiv2lib PRIVATE $<BUILD_INTERFACE:${ZLIB_LIBRARIES}>)
 endif()
 
 if( EXIV2_ENABLE_NLS )

--- a/xmpsdk/CMakeLists.txt
+++ b/xmpsdk/CMakeLists.txt
@@ -28,13 +28,12 @@ add_library(exiv2-xmp STATIC
 
 target_link_libraries(exiv2-xmp
     PRIVATE 
-        ${EXPAT_LIBRARY}
+        $<BUILD_INTERFACE:${EXPAT_LIBRARY}>
 )
 
 target_include_directories(exiv2-xmp
-    PUBLIC 
-        $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/xmpsdk/include>
     PRIVATE 
+        $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/xmpsdk/include>
         ${EXPAT_INCLUDE_DIR}
 )
 
@@ -50,8 +49,9 @@ if (BUILD_SHARED_LIBS)
     set_property(TARGET exiv2-xmp PROPERTY POSITION_INDEPENDENT_CODE ON)
 endif()
 
-# 1119  Install libxmp.a for use by third party applications (Thanks, Emmanuel)
-install(TARGETS exiv2-xmp EXPORT exiv2Config
-        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+# The private dependencies of a static library still need to be exported because they are needed to properly link the consumers of the static library.
+install(TARGETS exiv2-xmp EXPORT exiv2Config	
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}	
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}	
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
 )

